contracts/JuiceBuybackDelegate.sol

```diff
@@ -189,22 +171,19 @@ contract JuiceBuybackDelegate is IJBFundingCycleDataSource, IJBPayDelegate, IUni
 
       // If the amount minted is bigger than the lowest received when swapping, use the mint pathway
       if (_tokenCount >= _quote * _slippage / SLIPPAGE_DENOMINATOR) {
-        _mintedAmount = _tokenCount;
+        mintedAmount = _tokenCount;
+        reservedRate = _data.reservedRate;
```

Comment:
the if block is for minting & we are suing `reservedRate` while minting after the swap so this should be outside the if block 
wdyt ?

Reply:
Having it inside the if block allows to use the mutex only if we call the delegate later on (ie swap *looks* more interesting), it is then sloaded there (and mutex reset)